api/v1beta1/auth_config_types.go

```diff
@@ -774,6 +774,7 @@ func (s *AuthConfigStatus) Ready() bool {
 // AuthConfig is the schema for Authorino's AuthConfig API
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
+// +kubebuilder:storageversion
```

Comment:
shouldn't be v1beta2 the one to be stored?

Reply:
Not necessarily. Usually, first you introduce a new version of the API and allow time for users to adapt; then the newer version becomes the stored one in a subsequent rolling update. This way, when they install the new version of the CRD, previously existing CRs based on the old version won't be a blocker.

For ref: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#version-removal

Reply:
I am missing something. If the stored version is the old one, even the clients of the new version will store CR with the old version. `then the newer version becomes the stored one in a subsequent rolling update`. Then, there will be the same situation, all the CR stored versions would be the old one and the CRD's stored one will be the new one. I do not see any advantage of postponing the stored version. 

What I was assuming, and correct me if I am wrong, whenever a client updates /  creates an object, if the object's version is not the stored one, the webhook conversion API will be called and the object converted.

Reply:
Looking at the [Versions in CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/), the workflow for upgrading CRD version would be (high level):

* Release 1: Add a new version to the version list. The old version is the stored one. Both versions will be served. No deprecation flag.
* Release 2 (or later): Old version deprecation. The old version is served, but not stored and added deprecation warning. The new version is the stored (and served) one. Just after installing the new CRD with the changes, the stored (in old version) CR's need to be [migrated](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version). This is procedure needs to be run by some cluster admin. 
* Release 3 (or later): Version removal. When all the clients have migrated to the new version (one option is checking kube-apiserver logs), the old version can either be disabled for serving (`served: false`) or just removed from the CRD `spec.versions` list.

Reply:
Excellent summary, @eguzki. Thank you for this!